js/index.js

```diff
@@ -22,4 +23,50 @@ for (let i = 0; i < skills.length; i++) {
   skill.innerHTML = skills[i];
   //append 'skill' to 'skillsList'
   skillsList.appendChild(skill);
-}
\ No newline at end of file
+}
+
+// -- lesson 4-3
+//selection
+const messageForm = document.querySelector('[name=leave_message]')
+//event listener
+messageForm.addEventListener('submit', (Event) => {
+  //preventing default submit event
+  Event.preventDefault()
+  //form field variables retrieving input
+  const name = Event.target.name.value
+  console.log('name')
+  const email = Event.target.email.value
+  console.log('email')
+  const message = Event.target.message.value
+  console.log('message')
+  //selection
+  const messageSection = document.querySelector('#messages')
+  //selection
+  const messageList = messageSection.querySelector('ul')
+  //create
+  const newMessage = document.createElement('li')
+  //set 
+  newMessage.innerHTML = `<a href = "mailto:${email}">${name}</a> wrote: <span>${message}</span> `
+  //remove button
+  const removeButton = document.createElement('button')
+  //setting inner text
+  removeButton.innerText = 'remove'
+  //setting type attribute
+  removeButton.type = 'button'
+  //click event
+  removeButton.addEventListener('click', (Event) => {
+    //finding parent element
+    const entry = Event.target.parentNode
+    entry.remove()
+  })
+  //hide #messages when empty
+  if (messageList.children.length < 1) {
```

Comment:
Right now, after running this locally, I think this  actually causes the section to not display at messages all, even after submitting a message for the first time. When this block is commented out, the feature works again. I suspect you actually would want this in a listener for `remove`. 

To make this work as you are intending, I would suspect you would want to make the display `none` by default in index.html, and then add the following at line 45:
```
    messageSection.style.display = '';
```

Then move your `if` block here into the `remove` listener, with the appropriate adaptations so it picks up the relevant document elements needed, into scope. This isn't very DRY, but you could do something like:

```javascript
removeButton.addEventListener('click', (Event) => {
    //finding parent element
    const entry = Event.target.parentNode;
    entry.remove();
    //selection
    let msgSection = document.querySelector('#messages');
    //selection
    let msgList = msgSection.querySelector('ul');
    //hide #messages when empty
    if (msgList.children.length < 1) {
      msgSection.style.display = 'none'
    }

  })
```